pkg/controller/datavolume-controller.go

```diff
@@ -2601,3 +2638,261 @@ func GetRequiredSpace(filesystemOverhead float64, requestedSpace int64) int64 {
+	dv *cdiv1.DataVolume) *corev1.Pod {
+
+	workloadNodePlacement, err := GetWorkloadNodePlacement(r.client)
+	if err != nil {
+		return nil
+	}
+	// Generate individual specs
+	container := makeSizeDetectionContainerSpec(r.importerImage, r.pullPolicy)
+	volume := makeSizeDetectionVolumeSpec(sourcePvc.Name)
+	objectMeta := makeSizeDetectionObjectMeta(sourcePvc, dv)
+	// Assemble the pod
+	pod := &corev1.Pod{
+		ObjectMeta: *objectMeta,
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				*container,
+			},
+			Volumes: []corev1.Volume{
+				*volume,
+			},
+			RestartPolicy:     corev1.RestartPolicyOnFailure,
+			NodeSelector:      workloadNodePlacement.NodeSelector,
+			Tolerations:       workloadNodePlacement.Tolerations,
+			Affinity:          workloadNodePlacement.Affinity,
+			PriorityClassName: getPriorityClass(sourcePvc),
+		},
+	}
+
+	return pod
+}
+
+// makeSizeDetectionObjectMeta creates and returns the object metadata for the size-detection pod
+func makeSizeDetectionObjectMeta(sourcePvc *corev1.PersistentVolumeClaim, dataVolume *cdiv1.DataVolume) *metav1.ObjectMeta {
+	return &metav1.ObjectMeta{
+		Name:      sizeDetectionPodName(sourcePvc),
+		Namespace: sourcePvc.Namespace,
+		Labels: map[string]string{
+			common.CDILabelKey:       common.CDILabelValue,
+			common.CDIComponentLabel: common.ImporterPodName,
+		},
+		OwnerReferences: []metav1.OwnerReference{
+			*metav1.NewControllerRef(dataVolume, schema.GroupVersionKind{
+				Group:   cdiv1.SchemeGroupVersion.Group,
+				Version: cdiv1.SchemeGroupVersion.Version,
+				Kind:    "DataVolume",
+			}),
+		},
+	}
+}
+
+// makeSizeDetectionContainerSpec creates and returns the size-detection pod's Container spec
+func makeSizeDetectionContainerSpec(image, pullPolicy string) *corev1.Container {
+	return &corev1.Container{
+		Name:            "size-detection-volume",
+		Image:           image,
+		ImagePullPolicy: corev1.PullPolicy(pullPolicy),
+		Command:         []string{"/usr/bin/cdi-image-size-detection"},
+		Args:            []string{"-image-path", common.ImporterWritePath, "-termination-message-path", corev1.TerminationMessagePathDefault},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				MountPath: common.ImporterDataDir,
+				Name:      DataVolName,
+			},
+		},
+	}
+}
+
+// makeSizeDetectionVolumeSpec creates and returns the size-detection pod's Volume spec
+func makeSizeDetectionVolumeSpec(pvcName string) *corev1.Volume {
+	return &corev1.Volume{
+		Name: DataVolName,
+		VolumeSource: corev1.VolumeSource{
+			PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+				ClaimName: pvcName,
+			},
+		},
+	}
+}
+
+// inflateSizeWithOverhead inflates the obtained image size with proper overhead calculations
+func (r *DatavolumeReconciler) inflateSizeWithOverhead(imgSize int64, pvcSpec *corev1.PersistentVolumeClaimSpec) (resource.Quantity, error) {
+	var returnSize resource.Quantity
+
+	if resolveVolumeMode(pvcSpec.VolumeMode) == corev1.PersistentVolumeFilesystem {
```

Comment:
I think this can really reuse the function `volumeSize`. It is almost like a copy paste for some non trivial logic. 

```
func volumeSize(c client.Client, storage *cdiv1.StorageSpec, volumeMode *v1.PersistentVolumeMode) (*resource.Quantity, error) {

```